pages/profile/index.js

```diff
@@ -27,29 +33,75 @@ export default function ProfilePage() {
     return wasSuccsessful;
   };
 
-  if (status === "loading" || (status === "authenticated" && !session)) {
+  if (
+    status === "loading" ||
+    (status === "authenticated" && !session) ||
+    isLoadingEvents ||
+    errorEvents
+  ) {
     return <Loading />;
   }
   const userInfo = formatedUserInfo(session);
 
+  const eventsCreatedByUser = events.filter(
+    (event) => event.createdBy === session.user._id
+  );
+
+  const attendedEventsIds = new Set();
+  let attendedEvents = [];
+
+  session.user.attendedEvents.forEach((id) => attendedEventsIds.add(id));
+  attendedEvents = events.filter((event) => attendedEventsIds.has(event._id));
+  const accordionSections = [
+    {
+      id: 0,
+      title: "Erstellte Events",
+      component: EventList,
+      componentsProps: { events: eventsCreatedByUser || [] },
+      counter: eventsCreatedByUser.length,
+      canOpen: eventsCreatedByUser.length !== 0,
+      isHighlighted: eventsCreatedByUser.length !== 0,
+      isOpen: false,
+    },
+    {
+      id: 1,
+      title: "Meine Merkliste",
+      component: EventList,
+      componentsProps: { events: attendedEvents || [] },
+      counter: attendedEvents.length,
+      canOpen: attendedEvents.length !== 0,
+      isHighlighted: attendedEvents.length !== 0,
+      isOpen: false,
+    },
+  ];
+  if (openSection !== undefined && openSection < accordionSections.length) {
+    accordionSections[openSection].isOpen = true;
+  }
+
   if (!session) {
     return (
       <MessageCard>
         <p>Du bist nicht berechtigt diese Seite zu besuchen.</p>
         <ArrowButton onClick={() => signIn()}>Jetzt Anmelden</ArrowButton>
       </MessageCard>
     );
-  } else if (!editMode) {
-    return (
-      <Profile toggleEditMode={toggleEditMode} userInfo={userInfo}></Profile>
-    );
-  } else if (editMode) {
+  } else {
     return (
-      <ProfileForm
-        toggleEditMode={toggleEditMode}
-        userInfo={userInfo}
-        handleSubmit={handleSubmit}
-      ></ProfileForm>
+      <>
+        {editMode ? (
+          <ProfileForm
+            toggleEditMode={toggleEditMode}
+            userInfo={userInfo}
+            handleSubmit={handleSubmit}
+          ></ProfileForm>
+        ) : (
+          <Profile
+            toggleEditMode={toggleEditMode}
+            userInfo={userInfo}
+          ></Profile>
```

Comment:
This can also be self-closing:

```suggestion
          <Profile
            toggleEditMode={toggleEditMode}
            userInfo={userInfo}
          />
```